docs/06/01/faq.md

```diff
@@ -0,0 +1,177 @@
+import type { DatePickerProps as SDKDatePickerProps } from '@gusto/embedded-react-sdk'
+
+const MyCustomDatePicker = ({
+  label,
+  description,
+  errorMessage,
+  isRequired,
+  isDisabled,
+  isInvalid,
+  id,
+  name,
+  value,
+  onChange,
+  minDate,
+  maxDate,
+  ...props
+}: SDKDatePickerProps) => {
+  return (
+    <div className="date-field">
+      <label htmlFor={id || name}>
+        {label}
+        {isRequired && <span aria-hidden="true"> *</span>}
+      </label>
+      {description && <div className="description">{description}</div>}
+      <DatePicker
+        id={id || name}
+        selected={value}
+        onChange={date => onChange && onChange(date)}
+        disabled={isDisabled}
+        minDate={minDate}
+        maxDate={maxDate}
+        className={isInvalid ? 'invalid' : ''}
+      />
+      {isInvalid && errorMessage && <div className="error">{errorMessage}</div>}
+    </div>
+  )
+}
+```
+
+To understand the expected behavior of complex components, refer to the SDK's implementations:
+
+- [ComboBox on GitHub](https://github.com/Gusto/embedded-react-sdk/tree/main/src/components/Common/UI/ComboBox)
+- [DatePicker on GitHub](https://github.com/Gusto/embedded-react-sdk/tree/main/src/components/Common/UI/DatePicker)
+
+#### Can I use a different UI framework for my custom components?
+
+Yes, you can use any UI framework or library for your custom components, as long as they correctly implement the required props and behaviors. For example, you could use Material UI, Chakra UI, or any other React-compatible UI library.
+
+The only requirement is that your component adapter implements the `ComponentsContextType` interface.
+
+#### My custom component isn't working correctly. What should I check?
+
+1. **Ensure you've implemented all required props**: Each component has a specific set of props that it expects to receive and use. Make sure your component handles all of these props correctly.
+
+2. **Check event handlers**: Pay special attention to event handlers like `onChange` and `onBlur`. The SDK expects these to be called with specific parameters.
+
+3. **Verify accessibility**: The SDK's default components are built with accessibility in mind. Ensure your custom components maintain these accessibility features.
+
+4. **Use the right types**: Make sure you're importing and using the correct prop types from the SDK.
+
+5. **Debug with React DevTools**: Use React DevTools to inspect the props being passed to your components and compare with what you're expecting.
+
+### Troubleshooting
+
+#### I'm getting errors about missing components. What's wrong?
+
+If you're seeing errors about missing components, it's likely that you haven't provided an implementation for all the components the SDK is trying to use. Make sure your component adapter includes all the components used by the SDK, or use the `defaultComponents` to provide fallbacks.
+
+```tsx
+import { defaultComponents } from '@gusto/embedded-react-sdk'
```

Comment:
Similar to above, wondering if we should be recommending them to use GustoProvider if they want to use our defaultComponents since the GustoProvider provides the fallback components from defaultComponents automatically

Reply:
Updated